File: core/tmp/Python/refer-offer/lian_xu_zi_shu_zu_de_zui_da_he_lcof.py
from typing import *

def maxSubArray(nums: List[int]) -> int:
    if len(nums) == 0:
        return 0
    
    res = nums[0]
    for i in range(1, len(nums)):
        if nums[i] + nums[i - 1] > nums[i]:
            nums[i] = nums[i] + nums[i - 1]
        res = max(nums[i], res)
    
    return res

if __name__ == "__main__":
    print(maxSubArray([1]))

File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/LianXuZiShuZuDeZuiDaHeLcof.java
package com.dbc.algorithm.ReferOffer;

public class LianXuZiShuZuDeZuiDaHeLcof {
    public int maxSubArray(int[] nums) {
        int res = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int temp = nums[i] + nums[i - 1];
            if (temp > nums[i]) {
                nums[i] = temp;
            }
            res = Math.max(res, nums[i]);
        }
        return res;
    }
}
